dev-tools/check_project.sh

```diff
@@ -32,11 +32,36 @@ if [[ ${EUID} -eq 0 ]]; then
   echo -e "\\n${RED}""Running this script as root is not supported""${NC}\\n"
 fi
 
+declare -A lastCheckedCache
+declare -A updatedLastCheckedCache
+
+FAST_EXECUTION=$([[ "$*" == *"--fast"* ]] && echo true || echo false)
+
+# Load cache
+CACHE_FILE=".last_checked"
+if [ -f "$CACHE_FILE" ] && $FAST_EXECUTION; then
+  # shellcheck source=/dev/null
+  source -- "$CACHE_FILE" || rm "$CACHE_FILE"
+  "${updatedLastCheckedCache[@]@A}" && declare -A lastCheckedCache="${_#*=}"
```

Comment:
```suggestion
  "${UPDATED_LAST_CHECKED_CACHE[@]@A}" && LAST_CHECKED_CACHE="${_#*=}"
```
Not sure why we declare it again since it's global anyway

Reply:
The `declare -A` is needed here, as a string (on the RHS) is assigned to the variable `LAST_CHECKED_CACHE`, which is an associative array. Otherwise `shellcheck` is not happy

To me, this looked more sensible than to disable the check (and thus the warning).

```txt
In ./dev-tools/check_project.sh line 45:
  "${UPDATED_LAST_CHECKED_CACHE[@]@A}" && LAST_CHECKED_CACHE="${_#*=}"
                                          ^----------------^ SC2178 (warning): Variable was used as an array but is now assigned a string.

For more information:
  https://www.shellcheck.net/wiki/SC2178 -- Variable was used as an array but...

==> FIX ERRORS

```